src/annotator/guest.js

```diff
@@ -192,6 +192,7 @@ export class Guest {
     this._portFinder = new PortFinder({
       hostFrame: this._hostFrame,
       source: 'guest',
+      sourceId: this._frameIdentifier ?? undefined,
```

Comment:
> However, does this get set by the VitalSourceInjector?

VitalSourceInjector calls the `injectClient` function which adds a [`subFrameIdentifier` property to the client configuration](https://github.com/hypothesis/client/blob/f5760defbab8e26e72f9e91860bef6db085a5e18/src/annotator/hypothesis-injector.js#L98).  This config property is then saved to `this._frameIdentifier` in the guest.

Currently the value of this property in VS is a randomly generated string that changes each time a new chapter is loaded. My plan in future is to use a fixed identifier for the VS chapter content frame. This will allow the sidebar to know that the new guest frame is replacing the old one, rather than adding a new, unrelated guest.